### PR TITLE
[mlir][nvgpu] Simplify TMA IR generation

### PR DESCRIPTION
**Pull request by grypp**
This PR add `TmaDescriptorBuilder` 
- class simplifies TMA generation.
- Makes the code ready to support various Tma configurations 
   - removes strings and use the enums from `mlir.nvgpu.ENUMs`.
   - Example "swizzle = swizzle_128b, l2promo=none, oob=zero, interleave=none" to  enums in `mlir.nvgpu` dialects. 
   - Enums have string equivalent that are used during the IR writing and generation (see `TmaDescriptorBuilder::tensormap_descriptor_ty`). 
 - Improves readability and abstracts out TMA descriptor builders in reusable component.

### CODE REVIEW

**Review by manishucsd — APPROVED** [review]
LGTM.

File: mlir/test/Integration/GPU/CUDA/sm90/python/tools/matmulBuilder.py [hunk 1]
```diff
@@ -28,6 +28,43 @@
...
+    @property
+    def tensormap_descriptor_ty(self):
+        """Returns a tensormap descriptor type."""
+        memref_str = f"memref<{self.tma_box_shape[0]}x{self.tma_box_shape[1]}x{self.memref_ty.element_type}, 3>"
+        parse_str = f"!nvgpu.tensormap.descriptor<tensor = {memref_str},\
+                                              swizzle = {self.swizzle},\
+                                              l2promo = {self.l2promo},\
+                                              oob = {self.oob},\
+                                              interleave = {self.interleave}>"
+        return ir.Type.parse(parse_str)
```
Comment (jpienaar) [inline mlir/test/Integration/GPU/CUDA/sm90/python/tools/matmulBuilder.py:51]: Could we avoid parsing here and add to dialect as other dialects have done?

Comment (makslevental) [reply mlir/test/Integration/GPU/CUDA/sm90/python/tools/matmulBuilder.py:51]: extra :+1: from me on using [mlir_type_subclass](https://github.com/llvm/llvm-project/blob/91f11611337dde9a8e0a5e19240f6bb4671922c6/mlir/include/mlir/Bindings/Python/PybindAdaptors.h#L440) for this (example use [here](https://github.com/llvm/llvm-project/blob/681eacc1b670fd7137d8677fef6fc76c6e37dca9/mlir/lib/Bindings/Python/DialectTransform.cpp#L45))

Comment (grypp) [reply mlir/test/Integration/GPU/CUDA/sm90/python/tools/matmulBuilder.py:51]: Thanks for the pointers. Let me figure out how to do that.

Comment (makslevental) [reply mlir/test/Integration/GPU/CUDA/sm90/python/tools/matmulBuilder.py:51]: @grypp I pinged you on discord so feel free to reach out if you need/want help wiring it up.

**Pull request merged**